### DISCUSSION TITLE
how to make sure that all the faces created in T-model is a planar?

### DISCUSSION THREAD

**Discussion by mqc007** [neperfepx/neper#478] [Q&A] [answered]
Hi,
 
I met a problem in simulate, when I import the tess into 3dec, 3dec warns me the some faces are not a planar, I watched the guidebook in 3dec, it said that in the large strain situation the not planar faces might be a problem.

So is there any possible to make sure all the faces created are planar?

thanks in advance
all regards

**Comment by rquey** [comment, marked as answer]
Faces can become non planar as a result of regularization (`-reg 1`). To get strictly planar faces, disable regularization (`-reg 0`), or, to get "more planar" faces, leave regularization on but decrease the "flatness fault" tolerance (`-maxff`, 20° by default).

**Reply by mqc007** [reply]
thank you so much!

**Reply by ap264** [reply]
Hello mqc007

May I know how to import the tess file into 3dec?

Thank you
